aten/src/ATen/core/boxing/impl/make_boxed_from_unboxed_functor.h

```diff
@@ -204,6 +200,14 @@ namespace impl {
     static_assert(guts::false_t<T>::value,
       "You tried to register a kernel with an unsupported integral input type. Please use int64_t instead.");
   };
+  template<class T, bool AllowDeprecatedTypes>
+  struct assert_is_valid_input_type<T, AllowDeprecatedTypes, std::enable_if_t<std::is_same<const c10::SymInt&, T>::value>> {
+    static_assert(guts::false_t<T>::value,
+      "You tried to register a kernel taking c10::SymInt by reference. Please accept it by value instead.");
```

Comment:
Is this a compile-time test?

Reply:
Yes, compile time. But oddly, this doesn't seem to work for some reason... see the test I just posted.